common/sshMaxArg.py

```diff
@@ -22,56 +22,92 @@
-        ssh = config.sshCommand(cmd = ['printf', randomId(mid)],
-                                 nice = False,
-                                 ionice = False,
-                                 prefix = False)
+        ssh = config.sshCommand(
+            cmd=['printf', randomId(mid)],
+            nice=False,
+            ionice=False,
+            prefix=False)
+
         try:
             proc = subprocess.Popen(ssh,
-                                    stdout = subprocess.PIPE,
-                                    stderr = subprocess.PIPE,
-                                    universal_newlines = True)
+                                    stdout=subprocess.PIPE,
+                                    stderr=subprocess.PIPE,
+                                    universal_newlines=True)
             out, err = proc.communicate()
+
         except OSError as e:
+
             if e.errno == 7:
                 reportTest(mid, 'python exception: %s' % e.strerror)
                 mid -= r
+
             else:
                 raise
+
         else:
             l = len(out)
+
             if l == mid:
                 reportTest(mid, 'can be longer')
                 mid += r
+
             else:
                 reportTest(mid, 'is to long')
                 mid -= r
+
         r = round(r / 2)
+
     return mid + 6
 
+
 def reportTest(mid, msg):
+    """
+    """
     print('Check length %s:\t%s' % (mid, msg))
 
+
 def reportResult(host, mid):
+    """
+    """
     print('Maximum SSH argument length between "%s" and "%s" is %s'
           % (socket.gethostname(), host, mid))
 
-if __name__ == '__main__':
-    parser = argparse.ArgumentParser(description = 'Check maximal argument length on SSH connection',
-                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
-    parser.add_argument('MID',
-                        type = int,
-                        nargs = '?',
-                        default = 1048320,
-                        help = 'Start checking with MID arg length')
-    args = parser.parse_args()
-    import config
-    cfg = config.Config()
-    mid = maxArgLength(cfg, args.MID)
-    reportResult(cfg.sshHost(), mid)
+
+# buhtz: Couldn't find a hint that this script is executed diretly like
+# "./sshMaxArg.py". It more seems that is done via "import sshMaxArg"
+
+# if __name__ == '__main__':
+#     """
+#     """
+#     parser = argparse.ArgumentParser(
+#         description='Check maximal argument length on SSH connection',
+#         formatter_class=argparse.ArgumentDefaultsHelpFormatter
+#     )
+
+#     parser.add_argument('MID',
+#                         type=int,
+#                         nargs='?',
+#                         default=1048320,
+#                         help='Start checking with MID arg length')
+
+#     args = parser.parse_args()
+
+#     import config
+
+#     cfg = config.Config()
+#     mid = maxArgLength(cfg, args.MID)
+
+#     reportResult(cfg.sshHost(), mid)
```

Comment:
Can we be reasonably sure that this commenting-out doesn't break anything?

Reply:
I check the whole code via "grep" and searched for `subprocess` based calls of `sssMaxArg.py`.
I also checked the start scripts in the bin folder.
I couldn't find anything.

I assume that Germar or someone else had the idea first to implement that as an external tool and call it via `subprocess` when needed. But then it turned out to be usable directly via `import sshMaxArg`. But the `__main__` was left here.

We keep the code as a comment and not deleting it.

There are a lot of other files having there own `__main__` which makes not much sense in the current days. For example the unittest files. I think someone just wanted to run only one test file directly via `python test_foo.py`. With packages like `unittest` and `pytest` there is no need for this anymore.

Reply:
Okay, I understand. Let's go ahead with that.

Before any next release, we should absolutely test this in a real-world sshfs setup to make sure (not just rely on unit tests).

Reply:
Agree. I still collecting real-world-scenarios on my private todo-list. Because I use BIT only over SSH without something special.

Reply:
The main function servers a (debugging and issue testing) purpose documented here:

https://github.com/bit-team/backintime/blob/b14a98fc8aa24c6e51be0c7cd167b24a298da28a/common/config.py#L552-L555

Reply:
But I couldn't find a hint that `sshMaxArg.py` is every used like this.

The arg-length checks are done but via `import sshMaxArg`. Of course that ssh-marx-arg-thing also is an Issue for further investigation and refactoring. I was surprised to here that there is an argument limit. Maybe on older systems only?

Reply:
> But I couldn't find a hint that `sshMaxArg.py` is every used like this.

Neither did I and I think this is only done manually at the console by a human to diagnose SSH backup issues.
So I would keep this main function and put `python3 sshMaxArg.py <user@ssh-host>` on a checklist to diagnose SSH problems in issues (I didn't know until now that this functionality exists).